server/eclipse-project/src/main/java/us/freeandfair/corla/asm/ASMTransitionFunction.java

```diff
@@ -169,51 +168,69 @@ public ASMTransition value() {
-    D(new ASMTransition(AUDIT_INITIAL_STATE,
-                        NO_CONTESTS_TO_AUDIT_EVENT,
-                        AUDIT_REPORT_SUBMITTED)),
-    E(new ASMTransition(AUDIT_IN_PROGRESS,
+    F(new ASMTransition(WAITING_FOR_ROUND_START,
+                        ROUND_START_EVENT,
+                        ROUND_IN_PROGRESS)),
+    G(new ASMTransition(WAITING_FOR_ROUND_START,
+                        SIGN_OUT_AUDIT_BOARD_EVENT,
+                        WAITING_FOR_ROUND_START_NO_AUDIT_BOARD)),
+    H(new ASMTransition(WAITING_FOR_ROUND_START,
+                        RISK_LIMIT_ACHIEVED_EVENT,
+                        AUDIT_COMPLETE)),
+    M(new ASMTransition(ROUND_IN_PROGRESS,
+                        SetCreator.setOf(REPORT_MARKINGS_EVENT,
+                                         REPORT_BALLOT_NOT_FOUND_EVENT,
+                                         SUBMIT_AUDIT_INVESTIGATION_REPORT_EVENT),
+                        ROUND_IN_PROGRESS)),
+    N(new ASMTransition(ROUND_IN_PROGRESS,
                         SIGN_OUT_AUDIT_BOARD_EVENT,
-                        AUDIT_BOARD_SIGNED_OUT)),
-    F(new ASMTransition(AUDIT_IN_PROGRESS, 
-                        REPORT_MARKINGS_EVENT,
-                        AUDIT_IN_PROGRESS)),
-    G(new ASMTransition(AUDIT_IN_PROGRESS,
-                        REPORT_BALLOT_NOT_FOUND_EVENT,
-                        AUDIT_IN_PROGRESS)),
-    H(new ASMTransition(AUDIT_IN_PROGRESS,
-                        SUBMIT_AUDIT_INVESTIGATION_REPORT_EVENT,
-                        AUDIT_IN_PROGRESS)),
-    I(new ASMTransition(AUDIT_IN_PROGRESS,
-                        SUBMIT_INTERMEDIATE_AUDIT_REPORT_EVENT,
-                        AUDIT_BOARD_SIGNED_OUT)),
-    J(new ASMTransition(AUDIT_IN_PROGRESS,
+                        ROUND_IN_PROGRESS_NO_AUDIT_BOARD)),
+    O(new ASMTransition(ROUND_IN_PROGRESS,
                         ROUND_COMPLETE_EVENT,
                         WAITING_FOR_ROUND_SIGN_OFF)),
-    K(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF,
-                        ROUND_SIGN_OFF_EVENT,
-                        AUDIT_IN_PROGRESS)),
-    L(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF,
+    /* We probably want this transition eventually, but not for CDOS
+    EARLY(new ASMTransition(ROUND_IN_PROGRESS,
+                            RISK_LIMIT_ACHIEVED_EVENT, 
+                            AUDIT_COMPLETE)), */
+    P(new ASMTransition(ROUND_IN_PROGRESS_NO_AUDIT_BOARD,
+                        SIGN_IN_AUDIT_BOARD_EVENT,
+                        ROUND_IN_PROGRESS)),
+    Q(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF,
                         SIGN_OUT_AUDIT_BOARD_EVENT,
                         WAITING_FOR_ROUND_SIGN_OFF_NO_AUDIT_BOARD)),
-    M(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF_NO_AUDIT_BOARD,
+    R(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF,
+                        ROUND_SIGN_OFF_EVENT,
+                        WAITING_FOR_ROUND_START)),
+    S(new ASMTransition(WAITING_FOR_ROUND_SIGN_OFF,
```

Comment:
Does this allow a situation where a county X completed a round, county Y completes a round and signs off, hitting the risk limit, and then county X doesn't get a chance to sign off?  Or is the ASM such that we can't achieve the risk limit if some county is in the `WAITING_FOR_ROUND_SIGN_OFF` state?

Reply:
Remember that there are multiple audit board ASMs, one for each county. The risk limit being achieved is per-county, not per-contest or systemwide.

Reply:
(so, the answer to your question is, "no, the situation you describe cannot happen")